pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -247,6 +259,43 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	return nil
 }
 
+func finalizeIfNeeded(ctx context.Context, stream *imagev1.ImageStream, srcClusterName string, clients map[string]ctrlruntimeclient.Client) error {
+	finalizerSet := sets.NewString(stream.ObjectMeta.Finalizers...)
+	if finalizerSet.Has(ControllerName) {
+		for clusterName, client := range clients {
+			if clusterName == srcClusterName {
+				continue
+			}
+			// populate deleting to all clusters except srcClusterName
+			if err := client.Delete(ctx, &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      stream.Name,
+					Namespace: stream.Namespace,
+				},
+			}); !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+		// remove our finalizer from the list and update it.
+		stream.ObjectMeta.Finalizers = finalizerSet.Difference(sets.NewString(ControllerName)).List()
```

Comment:
This needs to be `finalizerSet.Delete(ControllerName).List()`

Reply:
GoLand did not show me `Delete` in the auto-completion.
It was too late for my laptop too.